### PR TITLE
Helperizing attestations, closes #827

### LINKED ISSUE #827: ENH: attestations -> helper
implement simon's attestations.py as a helper

### PR DESCRIPTION
**Pull request by pliu2781**
closes #827

### CODE REVIEW

**Comment by sbillinge** [comment]
@pliu2781 is there progress on this?

**Comment by pliu2781** [comment]
> @pliu2781 is there progress on this?

Yes, I am still sorting things out with the tests, because I had to add to the exemplars in schemas.py

**Comment by pliu2781** [comment]
closes #827

**Comment by sbillinge** [comment]
> 
> 
> closes #827

for everything to work properly, I believe that this should be added to the very first comment/description.  You can click ... and hit edit and and copy-paste this there.

File: regolith/helpers/attestationshelper.py [hunk 1]
```diff
@@ -123,21 +122,19 @@ def db_updater(self):
...
-                                ed = get_dates(p_appt).get("end_date")
-                                end = max(end, ed)
-                                months_on_grant = (ed - bd).days / 30.4 * loading
-                                appt = (person['_id'], bd, ed, loading, months_on_grant)
-                                appts.append(appt)
-            #                    on_grant = []
+                for _id, p_appt in person_appts.items():
+                    grantid = p_appt.get('grant')
+                    if grantid == grant_id:
+                        #accountnr = grant.get('account', grant['alias'])
```
Comment (sbillinge) [inline regolith/helpers/attestationshelper.py:128]: delete things we don't use.  You can do this at the end when we are sure we don't need it.

File: regolith/schemas.py [hunk 2]
```diff
@@ -231,6 +231,14 @@
             for i in range(1, 11)
         ],
         "payee": "scopatz",
+        "reimbursements": [
```
Comment (sbillinge) [inline regolith/schemas.py:234]: we should probably think about having a test for a non tbd reimbursement to cover everything, but decide if that is too much work and is worth it.

File: tests/test_helpers.py [hunk 3]
```diff
@@ -7,6 +7,33 @@
 
 dash = "-"
 helper_map = [
+    (["helper", "attestations", "dmref15", "-b", "2019-09-01", "-e", "2019-11-01", "--no-plot"],
```
Comment (sbillinge) [inline tests/test_helpers.py:10]: we will almost certainly have to pass in a date for testing, or the code will default to `today` on the day the test is run and the behavior will change over time.  Not sure, but that is usually the case.

**Review by sbillinge — COMMENTED** [review]
this looks great!  Please see few comments

File: regolith/schemas.py (hunk 2 above)
Comment (pliu2781) [reply regolith/schemas.py:234]: I added a non tbd reimbursement to test out

File: tests/test_helpers.py (hunk 3 above)
Comment (pliu2781) [reply tests/test_helpers.py:10]: If no begin/end_date is passed, the program just gets the begin/end_date from get_dates(grant)['begin/end_date']

**Comment by sbillinge** [comment]
Thanks so much, this is great!  I will check it out and test!

**Pull request merged**